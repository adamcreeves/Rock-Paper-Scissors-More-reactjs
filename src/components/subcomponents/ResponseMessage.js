import React from "react";
import { availableOptions } from "../../utils/Utils";
import "../../styles/Main.css";
import { str_015, str_016, str_020 } from "../../resources/strings";
import { int_004 } from "../../resources/integers";
import { c_011, c_012, c_013 } from "../../resources/classNames";

function ResponseMessage({
  player1,
  player2,
  player1sThrow,
  player2sThrow,
  winnerMessage,
}) {
  return (
    <div>
      {player2sThrow === int_004 ? (
        <div className={c_011}>
          <div className={c_012}>{str_020}</div>
        </div>
      ) : (
        <div className={c_011}>
          <div className={c_013}>
            {player1 +
              str_015 +
              availableOptions[player1sThrow] +
              str_016 +
              player2 +
              str_015 +
              availableOptions[player2sThrow]}
          </div>
          <div className={c_012}>{winnerMessage}</div>
        </div>
      )}
    </div>
  );
}

export default ResponseMessage;
