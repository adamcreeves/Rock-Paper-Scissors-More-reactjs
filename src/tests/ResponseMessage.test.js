import React from "react";
import { render } from "@testing-library/react";
import ResponseMessage from "../components/subcomponents/ResponseMessage";

describe("Test Body", () => {
  it("should match the snapshot", () => {
    const { asFragment } = render(<ResponseMessage />);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should render Body and not be null", () => {
    const { asFragment } = render(<ResponseMessage />);
    expect(asFragment()).not.toBeNull();
  });
});
