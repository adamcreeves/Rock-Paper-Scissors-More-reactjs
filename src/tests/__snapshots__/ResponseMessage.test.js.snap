// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Body should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="responseMessage"
    >
      <div
        class="responseMessage__throws"
      >
        undefined throws undefined and undefined throws undefined
      </div>
      <div
        class="responseMessage__winner"
      />
    </div>
  </div>
</DocumentFragment>
`;
